app/javascript/packs/custom.js

```diff
@@ -22,4 +22,37 @@ $(document).ready(function(){
         });
         return false
     });
+
+    let cart = $(".cart-modal");
+    $("#btn-cart").click(function(event) {
+        event.preventDefault();
+        if (cart.hasClass("animation") || cart.hasClass("reverse-animation"))
+            return;
+        if (cart.hasClass("hide")){
+            cart.addClass("animation").removeClass("hide");
+            setTimeout(function(){
+                cart.removeClass("animation")
+            }, 1500);
+        } else {
+            cart.addClass("reverse-animation");
+            setTimeout(function () {
```

Comment:
if you not use $(this), please, use arrow functions.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions